Add document for Dreo integration (#36851)
home-assistant/home-assistant.github.io: source/_integrations/dreo.markdown

---
title: Dreo
description: Instructions on how to set up Dreo products within Home Assistant.
ha_category:
  - Fan
ha_release: 2024.12
ha_domain: dreo
ha_platforms:
  - fan
ha_integration_type: integration
---

> The **Dreo** {% term integration %} enables you to control smart products connected to the [Dreo App](https://m.dreo.com/en).

Reviewer: should we also add a link to https://www.dreo.com/ ?

The devices must be added to the Dreo App before this integration can discover them.

The following platforms are supported:

- **fan**

## Supported devices

This integration supports devices controllable by the Dreo App. The following devices are supported by this integration:

### Fans

> #### General Features

Reviewer: I think this section should focus more on what the integration provides, and not on the product features. For product features users can check the manufacturers website/docs.
I would also suggest making the sentences a bit simpler and direct.

See https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/docs-supported-functions/

> - **Sleep Mode**: All models can adjust the fan speed smoothly in the "sleep" mode. However, some models (such as DR-HTF001S) perform better in terms of adjustment smoothness, providing a more stable and quiet environment for sleeping.

Reviewer: is this part of `preset_modes`?

Author: @abmantis Yes, sleep mode is implemented as part of the preset_modes functionality. In the integration, users can select 'sleep' as one of the preset modes for the fan. I've updated the documentation description to make this clearer by explicitly stating that it's a "sleep" preset mode rather than just a "sleep" mode.

Author: @abmantis  I've updated the documentation according to your suggestion, placing the "## Removing the integration" heading before the include statement. Thank you for your helpful feedback to improve the documentation structure.

> - **Wide-Angle Oscillation**: Fans are typically equipped with a wide oscillation function, which ensures more even air circulation—especially in larger spaces.

Reviewer: ```suggestion
- **Wide-Angle Oscillation**: Fans are typically equipped with a wide oscillation function, which ensures more even air circulation — especially in larger spaces.
```

- **Intelligent Auto Mode**: The "auto" mode of each model can adjust the fan speed based on temperature and humidity. For example, it provides a more precise and personalized comfort experience.

> - **Child Lock Function**: Many fans include a child lock that helps prevent accidental setting changes and enhances household safety.

Reviewer: child lock is not implemented

Author: @abmantis  You're absolutely right. Thank you for pointing this out. I've removed the Child Lock Function from the features list since it's not implemented in the integration.

- **Fast Speed Adjustment Response**: Most fans offer a rapid speed-adjustment response time, enabling you to swiftly alter the fan speed according to your requirements.

> - **Multi-Voice Assistant Integration**: Many models support integration with a wider range of voice assistants, which facilitates a wider range of hands-free operations.

Reviewer: ```suggestion
```
this is not specific to this integration

Author: @abmantis I've removed the Multi-Voice Assistant Integration line from the features list as suggested. This change helps ensure the documentation accurately reflects the functionality available in the Home Assistant integration for Dreo devices.

> - **Energy-Efficient Motor**: Many fans come with an energy-efficient motor that reduces electricity costs while delivering strong airflow.

Reviewer: ```suggestion
```

Author: @abmantis The above issue has already been resolved.

> - **Long Timer Setting**: Most fans offer a timer setting of up to 12 hours, which gives you more flexibility in scheduling the fan's operation.

Reviewer: ```suggestion
```

Author: @abmantis The above issue has already been resolved.

The supported models include: DR-HTF001S, DR-HTF002S, DR-HTF005S, DR-HTF007S, DR-HTF008S, DR-HTF009S, DR-HTF010S.

#### Model-Specific Features

> - **Wide-Angle Oscillation**: Generally, fans are equipped with a wide-angle oscillation function. However, the fan of model DR-HTF004S distinguishes itself with its remarkable performance. It offers an oscillation range of up to 120 degrees, and its oscillation mechanism is precisely engineered to ensure a more uniform and efficient air circulation. This feature is particularly advantageous in larger spaces, effectively spreading fresh air across the area.

Reviewer: there isn't anything different in the way the integration handles this fan or how it is presented in Home Assistant, so I think we can remove this section

Author: @abmantis  I've removed the Model-Specific Features section as suggested. You're right th@at there's no difference in how the integration handles the DR-HTF004S model in Home Assistant compared to other models.

The supported models include: DR-HTF004S

## Prerequisite

Before you can use this integration, all devices must be registered with the
Dreo App. Once registration is complete, you can add the Dreo integration in Home Assistant through the UI.

{% include integrations/config_flow.md %}

> {% include integrations/remove_device_service.md %}

Reviewer: ```suggestion
## Removing the integration

{% include integrations/remove_device_service.md %}
```

## Fan exposed attributes

Dreo Tower Fan will expose the following details depending on the features supported by the model:

| Attribute | Description                                                  | Example |
| --------- | ------------------------------------------------------------ | ------- |
| `mode`    | The current mode the device is in. | "manual", "auto", "sleep" |
| `speed`   | The current speed setting (1-4). | 1 |
| `oscillate` | Whether oscillation is enabled. | true/false |

All attributes are supported by models: DR-HTF001S, DR-HTF002S, DR-HTF004S, DR-HTF005S, DR-HTF007S, DR-HTF008S, DR-HTF009S, DR-HTF010S

## Troubleshooting

### Device not discovered

- **Check Dreo App Connection**: Ensure that your devices are properly connected to the Dreo App. Try restarting the Dreo App and checking if the devices are still visible and controllable within it. If not, re-pair the devices with the Dreo App, following the app's official pairing instructions.

> - **Home Assistant Network Issues**: Make sure that your Home Assistant instance has network access. You can test this by trying to access other online services from within Home Assistant (for example, checking the weather integration if it's set up). If there are network problems, troubleshoot your Home Assistant's network connection, which may involve checking your router settings, Wi-Fi passwords, or Ethernet connections.
> - **Permissions**: Double-check that you've granted all the necessary permissions during the integration setup process. Incorrect permissions can prevent Home Assistant from discovering your Dreo devices.

Reviewer: ```suggestion
```
these are not specific to the integration

Author: @abmantis I've removed the "Home Assistant Network Issues" and "Permissions" troubleshooting points as they're not specific to the Dreo integration but rather general issues applicable to most integrations.

### Inability to control devices

> - **Device State Sync**: Sometimes, the device state may not sync correctly between the Dreo App and Home Assistant. Try toggling the device on/off, changing the mode, or adjusting the speed from within the Dreo App, and then refreshing the device page in Home Assistant to see if the state updates.

Reviewer: ```suggestion
- **Device State Sync**: Sometimes, the device state may not sync correctly between the Dreo App and Home Assistant. Try toggling the device on/off, changing the mode, or adjusting the speed from within the Dreo App to see if the state updates.
```
HA updates without the need to refresh the page

Author: @abmantis I've updated the Device State Sync troubleshooting text as suggested, removing the part about refreshing the device page in Home Assistant. This simplifies the instructions and focuses on actions in the Dreo App that might help resolve sync issues.

> - **App Version Mismatch**: Ensure that both the Dreo App and Home Assistant are running up-to-date versions. An outdated version of either software could lead to compatibility issues and control problems. Update the Dreo App from the official app store for your device, and check for Home Assistant updates in the Home Assistant UI's update section.

Reviewer: since HA connects to the backend and not the app, how does the app version impact HA?

Author: @abmantis I've made the requested change. I kept the App Version Mismatch section but removed the reference to Home Assistant updates. The focus is now only on keeping the Dreo App up-to-date to avoid compatibility issues and control problems.

### Dreo App credentials are invalid or expired

If your Dreo App credentials are invalid or expired, you can follow these steps:

1. Open the Dreo App. Navigate to the account settings section.
2. Click on the "Log out" option if you are currently logged in.
3. Enter your correct username and password. If you have forgotten your password, use the "Forgot password" option in the app to reset it.
4. Once you have successfully logged in to the Dreo App, go back to Home Assistant. Navigate to the Dreo integration settings and re-enter your Dreo App credentials. Then save the settings.

### Device becomes unresponsive

If the device becomes unresponsive, try the following steps:

1. **Power cycle the device**: Turn off the device using its power switch or unplug it from the power source. Wait for about 10–15 seconds and then turn it back on or plug it back in.
2. **Check the Wi-Fi connection**: Ensure that the device is connected to a stable Wi-Fi network. You can try moving the device closer to the router to improve the signal strength.

> 3. **Restart the Dreo App and Home Assistant**: Close the Dreo App completely on your mobile device and restart it. Also, restart your Home Assistant instance. Then try to control the device again.

Reviewer: again, HA connects to the backend, right? why would restarting the app impact anything?
also, why restart HA? that shouldn't be required.

Author: @abmantis Your are right,I've removed the step about restarting the Dreo App and Home Assistant from the troubleshooting section. Since Home Assistant connects directly to the Dreo backend (not the app), restarting the app wouldn't impact functionality. Similarly, restarting Home Assistant shouldn't be required to resolve device connectivity issues. The focus is now on addressing the actual device connection by power cycling, checking Wi-Fi, and if necessary, resetting the device.

4. **Reset the device**: If the above steps do not work, you may need to reset the device to its factory settings. Refer to the device's user manual for the specific reset procedure. After resetting, re-add the device to the Dreo App and then to Home Assistant.

> ## Firmware update scenarios
> 
> ### Updating the device firmware
> 
> 1. **Check for updates in the Dreo App**: Open the Dreo App and navigate to the device management section. Select the device you want to update. If there is a firmware update available, the app will prompt you.
> 2. **Start the update**: Follow the on-screen instructions in the Dreo App to start the firmware update process. Make sure the device is connected to a stable power source and Wi-Fi network during the update.
> 3. **Wait for the update to complete**: The update process may take several minutes. Do not turn off the device or interrupt the update during this time.
> 4. **Verify the update**: Once the update is complete, check if the device's features are working as expected. You can also check the device information in the Dreo App to confirm the new firmware version.
> 
> ### Issues during firmware update
> 
> - **Update fails**: If the firmware update fails, check your network connection. Try restarting the device, the Dreo App, and your Wi-Fi router. Then attempt the update again.
> - **Device becomes unresponsive after update**: If the device becomes unresponsive after the firmware update, perform a power cycle on the device. If the problem persists, reset the device to its factory settings and re-add it to the Dreo App and Home Assistant.

Reviewer: ```suggestion
```
I am not sure if these should be here, since the update process is not done trough HA

Author: @abmantis Our intention with adding the firmware update section was to guide users on how to update device firmware and troubleshoot issues that might arise during updates. Based on your suggestion that this content may not align with Home Assistant documentation standards, I've removed the entire firmware update section. This ensures the documentation remains focused on functionality and operations directly relevant to the Home Assistant integration.

---
title: Dreo

> description: Instructions on how to set up Dreo fans within Home Assistant.

Reviewer: ```suggestion
description: Instructions on how to set up Dreo products within Home Assistant.
```
So that this does not get outdated as soon as more devices are added. We already state that only fans are supported in the actual docs bellow.

Author: Thank you for your reminder. I have already changed "Fans" to "products".

ha_category:
  - Fan

> ha_release: 2024.12

Reviewer: This should be updated when the core PR is merged so that it matches the actual release

ha_domain: dreo
ha_platforms:
  - fan
ha_integration_type: integration
---

> The Dreo integration enables you to control smart switches connected to the Dreo App.

Reviewer: ```suggestion
The **Dreo** {% term integration %} enables you to control smart switches connected to the Dreo App.
```
Also add a link to the official website, please.

- **Energy-Efficient Motor**: Many fans come with an energy-efficient motor that reduces electricity costs while delivering strong airflow.
- **Long Timer Setting**: Most fans offer a timer setting of up to 12 hours, which gives you more flexibility in scheduling the fan's operation.

The supported models include: DR-HTF001S, DR-HTF002S, DR-HTF005S, DR-HTF007S, DR-HTF008S, DR-HTF009S, DR-HTF010S.

#### Model-Specific Features

- **Wide-Angle Oscillation**: Generally, fans are equipped with a wide-angle oscillation function. However, the fan of model DR-HTF004S distinguishes itself with its remarkable performance. It offers an oscillation range of up to 120 degrees, and its oscillation mechanism is precisely engineered to ensure a more uniform and efficient air circulation. This feature is particularly advantageous in larger spaces, effectively spreading fresh air across the area.

The supported models include: DR-HTF004S

## Prerequisite

Before you can use this integration, all devices must be registered with the
Dreo App. Once registration is complete, you can add the Dreo integration in Home Assistant through the UI.

> ## Configuration instructions

Reviewer: We provide helpers here to generate the docs for normal config flows. You can use `{% include integrations/config_flow.md %}` to avoid detailing the steps.

Same for removal with `{% include integrations/remove_device_service.md %}`

Author: I've fixed  it,thanks

### Fans

- **Fan**
  - **Description**: Control your Dreo fan device.
  - **Functionality**:
    - Turn on/off
    - Set fan speed
    - Toggle oscillation
    - Select preset modes
  - **Available for models**: DR-HTF001S, DR-HTF002S, DR-HTF004S, DR-HTF005S, DR-HTF007S, DR-HTF008S, DR-HTF009S, DR-HTF010S
  - **Attributes**:
    - `mode` - The current mode the device is in ("manual", "auto", "sleep")
    - `speed` - The current speed setting
    - `oscillate` - Whether oscillation is enabled (true/false)

## Troubleshooting

### Inability to control devices

- **Device State Sync**: Sometimes, the device state may not sync correctly between the Dreo App and Home Assistant. Try toggling the device on/off, changing the mode, or adjusting the speed from within the Dreo App to see if the state updates.
- **App Version Mismatch**: Ensure that the Dreo App is running an up-to-date version. An outdated version could lead to compatibility issues and control problems. Update the Dreo App from the official app store for your device.

### Dreo App credentials are invalid or expired

If your Dreo App credentials are invalid or expired, you can follow these steps:

> 1. Open the Dreo App. Navigate to the account settings section.

Reviewer 2: ```suggestion
1. Open the Dreo App. Go to the account settings section.
```

> 2. Click on the "Log out" option if you are currently logged in.

Reviewer 2: ```suggestion
2. If you are currently logged in, select the **Log out** option.
```
- We prefer [select over click](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/c/click).
- We use [bold to markup UI strings](https://developers.home-assistant.io/docs/documenting/general-style-guide).

> 3. Enter your correct username and password. If you have forgotten your password, use the "Forgot password" option in the app to reset it.

Reviewer 2: ```suggestion
3. Enter your correct username and password. If you have forgotten your password, use the **Forgot password** option in the app to reset it.
```

4. Once you have successfully logged in to the Dreo App, go back to Home Assistant. Navigate to the Dreo integration settings and re-enter your Dreo App credentials. Then save the settings.

### Device becomes unresponsive

If the device becomes unresponsive, try the following steps:

> 1. **Power cycle the device**: Turn off the device using its power switch or unplug it from the power source. Wait for about 10–15 seconds and then turn it back on or plug it back in.

Reviewer 2: ```suggestion
1. Power cycle the device: 
   - Turn off the device using its power switch or unplug it from the power source. 
   - Wait for about 10–15 seconds and then turn it back on or plug it back in.
```
We use [bold to markup UI strings](https://developers.home-assistant.io/docs/documenting/general-style-guide). We don't use it to highlight text.

> 2. **Check the Wi-Fi connection**: Ensure that the device is connected to a stable Wi-Fi network. You can try moving the device closer to the router to improve the signal strength.

Reviewer 2: ```suggestion
2. Check the Wi-Fi connection: 
   - Ensure that the device is connected to a stable Wi-Fi network. 
   - You can try moving the device closer to the router to improve the signal strength.
```

> 3. **Reset the device**: If the above steps do not work, you may need to reset the device to its factory settings. Refer to the device's user manual for the specific reset procedure. After resetting, re-add the device to the Dreo App and then to Home Assistant.

Reviewer 2: ```suggestion
3. Reset the device: 
   - If the above steps do not work, you may need to reset the device to its factory settings. 
   - Refer to the device's user manual for the specific reset procedure. 
   - After resetting, re-add the device to the Dreo App and then to Home Assistant.
```

---
title: Dreo
description: Instructions on how to set up Dreo products within Home Assistant.
ha_category:
  - Fan
ha_release: 2026.5
ha_domain: dreo
ha_platforms:
  - fan
ha_integration_type: integration
---

The **Dreo** {% term integration %} enables you to control smart products connected to the [Dreo App](https://m.dreo.com/en) from [Dreo](https://www.dreo.com/).

> 

Reviewer 2: ```suggestion
```

## Prerequisites

Before you can use this integration, all devices must be registered with the
Dreo App. Once registration is complete, you can add the Dreo integration in Home Assistant through the UI.

{% include integrations/config_flow.md %}

> 

Reviewer 2: ```suggestion
```

---
title: Dreo
description: Instructions on how to set up Dreo products within Home Assistant.
ha_category:
  - Fan
ha_release: 2024.12
ha_domain: dreo
ha_platforms:
  - fan
ha_integration_type: integration
---

The **Dreo** {% term integration %} enables you to control smart products connected to the [Dreo App](https://m.dreo.com/en) from [Dreo](https://www.dreo.com/).

The devices must be added to the Dreo App before this integration can discover them.

Supported platforms:

- **fan**

## Supported functionality

> ### Entities
> 
> The Dreo integration provides the following entities:
> 
> #### Fans
> 
> - **Fan**
>   - **Description**: Control your Dreo fan device.
>   - **Functionality**:
>     - Turn on/off
>     - Set fan speed
>     - Toggle oscillation
>     - Select preset modes
>   - **Available for models**: DR-HTF001S, DR-HTF002S, DR-HTF004S, DR-HTF005S, DR-HTF007S, DR-HTF008S, DR-HTF009S, DR-HTF010S
>   - **Attributes**:
>     - `mode` - The current mode the device is in ("manual", "auto", "sleep")
>     - `speed` - The current speed setting
>     - `oscillate` - Whether oscillation is enabled (true/false)

Reviewer 2: ```suggestion
```
I would move this section after the config section

## Prerequisites

Before you can use this integration, all devices must be registered with the
Dreo App. Once registration is complete, you can add the Dreo integration in Home Assistant through the UI.

> {% include integrations/config_flow.md %}

Reviewer 2: ```suggestion
{% include integrations/config_flow.md %}

## Supported functionality

The Dreo integration provides the following entities:

### Fans

- **Fan**
  - **Description**: Control your Dreo fan device.
  - **Functionality**:
    - Turn on/off
    - Set fan speed
    - Toggle oscillation
    - Select preset modes
  - **Available for models**: DR-HTF001S, DR-HTF002S, DR-HTF004S, DR-HTF005S, DR-HTF007S, DR-HTF008S, DR-HTF009S, DR-HTF010S
  - **Attributes**:
    - `mode` - The current mode the device is in ("manual", "auto", "sleep")
    - `speed` - The current speed setting
    - `oscillate` - Whether oscillation is enabled (true/false)
```

## Remove the integration

> {% include integrations/remove_device_service.md %}

Reviewer 2: ```suggestion
```
we usually keep this section at the end of the page.

### Dreo App credentials are invalid or expired

If your Dreo App credentials are invalid or expired, you can follow these steps:

1. Open the Dreo App. Navigate to the account settings section.
2. Click on the "Log out" option if you are currently logged in.
3. Enter your correct username and password. If you have forgotten your password, use the "Forgot password" option in the app to reset it.
4. Once you have successfully logged in to the Dreo App, go back to Home Assistant. Navigate to the Dreo integration settings and re-enter your Dreo App credentials. Then save the settings.

### Device becomes unresponsive

If the device becomes unresponsive, try the following steps:

1. **Power cycle the device**: Turn off the device using its power switch or unplug it from the power source. Wait for about 10–15 seconds and then turn it back on or plug it back in.
2. **Check the Wi-Fi connection**: Ensure that the device is connected to a stable Wi-Fi network. You can try moving the device closer to the router to improve the signal strength.

> 3. **Reset the device**: If the above steps do not work, you may need to reset the device to its factory settings. Refer to the device's user manual for the specific reset procedure. After resetting, re-add the device to the Dreo App and then to Home Assistant.

Reviewer 2: ```suggestion
3. **Reset the device**: If the above steps do not work, you may need to reset the device to its factory settings. Refer to the device's user manual for the specific reset procedure. After resetting, re-add the device to the Dreo App and then to Home Assistant.

## Remove the integration

This integration follows standard integration removal.

{% include integrations/remove_device_service.md %}
```

---
title: Dreo
description: Instructions on how to set up Dreo products within Home Assistant.
ha_category:
  - Fan

> ha_release: 2024.12

Reviewer 2: ```suggestion
ha_release: 2026.5
```

ha_domain: dreo
ha_platforms:
  - fan
ha_integration_type: integration
---

The **Dreo** {% term integration %} enables you to control smart products connected to the [Dreo App](https://m.dreo.com/en) from [Dreo](https://www.dreo.com/).

> The devices must be added to the Dreo App before this integration can discover them.

Reviewer 2: ```suggestion
```
Could this be removed here? I think it is mentioned under the prerequisites.

Supported platforms:

- **fan**

> 

Reviewer 2: ```suggestion
```
Could this be removed for now? I think it is already listed in the supported functionality section.

## Prerequisites

> Before you can use this integration, all devices must be registered with the

Reviewer 2: should the Dreo app be listed here as well?

Dreo App. Once registration is complete, you can add the Dreo integration in Home Assistant through the UI.

> 

Reviewer 2: is there anything else I need to have ready before adding the integration? Do I need to enter credentials during integration setup, for example?

1. Open the Dreo App. Navigate to the account settings section.
2. Click on the "Log out" option if you are currently logged in.
3. Enter your correct username and password. If you have forgotten your password, use the "Forgot password" option in the app to reset it.
4. Once you have successfully logged in to the Dreo App, go back to Home Assistant. Navigate to the Dreo integration settings and re-enter your Dreo App credentials. Then save the settings.

### Device becomes unresponsive

If the device becomes unresponsive, try the following steps:

1. **Power cycle the device**: Turn off the device using its power switch or unplug it from the power source. Wait for about 10–15 seconds and then turn it back on or plug it back in.
2. **Check the Wi-Fi connection**: Ensure that the device is connected to a stable Wi-Fi network. You can try moving the device closer to the router to improve the signal strength.
3. **Reset the device**: If the above steps do not work, you may need to reset the device to its factory settings. Refer to the device's user manual for the specific reset procedure. After resetting, re-add the device to the Dreo App and then to Home Assistant.

> ## Remove the integration

Reviewer 2: ```suggestion
## Removing the integration
```

---
title: Dreo
description: Instructions on how to set up Dreo products within Home Assistant.
ha_category:
  - Fan
ha_release: 2026.5

> ha_domain: dreo

Reviewer 2: ```suggestion
ha_domain: dreo
ha_codeowners:
  - '@w-xtao'
```

---
title: Dreo
description: Instructions on how to set up Dreo products within Home Assistant.
ha_category:
  - Fan
ha_release: 2026.5

> ha_domain: dreo

Reviewer 2: ```suggestion
ha_domain: dreo
ha_iot_class: Cloud Polling
```

---
title: Dreo
description: Instructions on how to set up Dreo products within Home Assistant.
ha_category:
  - Fan
ha_release: 2026.5
ha_domain: dreo
ha_platforms:
  - fan
ha_integration_type: integration
ha_config_flow: true
---

The **Dreo** {% term integration %} enables you to control smart products connected to the [Dreo App](https://m.dreo.com/en) from [Dreo](https://www.dreo.com/).

## Prerequisites

Before you can use this integration, all devices must be registered with the
Dreo App. Once registration is complete, you can add the Dreo integration in Home Assistant through the UI.

{% include integrations/config_flow.md %}

## Configuration parameters

> To add the Dreo integration, you need to provide the following credentials from your Dreo App account:

Reviewer 2: ```suggestion
```
